Add the automated utilization reporting doc (#20130)
hashicorp/vault: website/content/docs/configuration/reporting.mdx

---
layout: docs
page_title: Reporting - Configuration
description: |-
  The reporting stanza configures per-cluster automated reporting of billing
  information. Vault Enterprise will automatically report the cluster's client
  counts using the license start date. Any modification of the default behavior
  should be specified in the reporting stanza.
---

# `reporting` stanza

The reporting stanza specifies configuration for [Automated entitlement
utilization reporting](/vault/docs/enterprise/license/utilization-reporting). 

<CodeBlockConfig hideClipboard>

```hcl
reporting {
	license {
		...
	}
}
```

</CodeBlockConfig>

## Requirements

A valid Vault Enterprise license is required to enable automated reporting of

> billing information.

Author: Good point. I'll update this.

## `reporting` parameters

> The automated reporting feature exposes the following system settings:

Reviewer: ```suggestion
The automated reporting feature can be configured via the following parameters:
```

precedence over values in the configuration file.

<Note>

Vault Enterprise uses client counts to calculate its entitlement utilization. To
provide the most accurate information, client count configuration parameters
require some restrictions while reporting is enabled. For more information,
refer to the [client counts API
doc](/vault/api-docs/system/internal-counters/#update-the-client-count-configuration).

</Note>

## Configuring a billing start timestamp

Vault Enterprise defaults to reporting client counts periodically for the
current billing period. The current billing period must be specified via the
`reporting` stanza:

```hcl
reporting {
	license {
        billing_start_timestamp = "2022-03-01T00:00:00Z"
	}
}
```

If no `billing_start_timestamp` is specified, Vault Enterprise will use the date
associated with your Vault Enterprise license.

<Note>

Vault Enterprise will not replicate reporting configuration changes to other
clusters. Any changes to the `reporting` stanza must be specified within each
cluster's HCL.

</Note>

## Disabling automated reporting

Reporting of entitlement utilization is enabled by default and you can disable

> it using the `reporting` stanza. To disable the automated reporting, set the

Reviewer: ```suggestion
it using the `reporting` stanza. To disable automated reporting, set the
```
